### PR TITLE
feat: userEvent.tab

### PR DESCRIPTION
**Pull request by benmonro**
implements https://github.com/testing-library/user-event/issues/167

### CODE REVIEW

**Comment by benmonro** [comment]
@alexkrolick this should be good to go.  welcome your feedback.  I was on the fence about adding a warning for tabIndex w/ a positive value, but decided that there are lint rules for that, i could be convinced either way though.  let me know.

File: __tests__/react/tab.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
+import React from "react";
+import { render, cleanup } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+import userEvent from "../../src";
+
+afterEach(cleanup);
+
+describe("userEvent.tab", () => {
+    it("should go from one element to the next, then back to the 1st", () => {
```
Comment (alexkrolick) [inline __tests__/react/tab.js]:
```suggestion
     it("should cycle elements in document tab order", () => {
```

File: __tests__/react/tab.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+        expect(document.activeElement).toBe(radio);
+
+        userEvent.tab();
+
+        expect(document.activeElement).toBe(number);
+
+        userEvent.tab();
+
+        expect(document.activeElement).toBe(checkbox);
```
Comment (alexkrolick) [inline __tests__/react/tab.js]:
```suggestion
         // cycle goes back to first element
         expect(document.activeElement).toBe(checkbox); 
```

File: src/index.js [hunk 3]
```diff
@@ -224,6 +224,30 @@ const userEvent = {
...
+    });
+    const index = list.indexOf(document.activeElement);
+
+    let nextIndex = shift ? index - 1 : index + 1;
+    let defaultIndex = shift ? list.length - 1 : 0;
+
+    const next = list[nextIndex] || list[defaultIndex];
+
+    if (next.getAttribute("tabindex") === null) {
+      next.setAttribute("tabindex", "0"); // jsdom requires tabIndex=0 for an item to become 'document.activeElement' (the browser does not)
```
Comment (kentcdodds) [inline src/index.js]: Should we unset this after `next.focus()`? It would be good if this function could be as pure as possible.

Comment (benmonro) [reply src/index.js]: @kentcdodds good call, i'd thought about this but was probably driving and forgot to add it. lol

done

File: __tests__/react/tab.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
+import React from "react";
+import { render, cleanup } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+import userEvent from "../../src";
+
+afterEach(cleanup);
```
Comment (Gpx) [inline __tests__/react/tab.js]: This should not be needed anymore with the latest version of RTL

**Comment by Gpx** [comment]
Can you please rebase and use `npm commit` so that we can release a new version?

**Comment by benmonro** [comment]
@Gpx not sure what `npm commit` is, but afaict this should be good to merge.

**Comment by kentcdodds** [comment]
@gpx, if you use GitHub's rebase and merge feature then you can write the commit message yourself and not worry about the commit messages that people write themselves. Saves everyone a lot of time and you don't need commitizen 👍

**Pull request merged**